internal/generator/ecc/template/tests/point.go.tmpl

```diff
@@ -584,6 +584,51 @@ func Benchmark{{ $TJacobian }}IsInSubGroup(b *testing.B) {
 
 }
 
+func Benchmark{{ $TJacobian }}Equal(b *testing.B) {
+	var scalar {{ .CoordType}}
+	if _, err := scalar.SetRandom(); err != nil {
+		b.Fatalf("failed to set scalar: %s", err)
+	}
+
+	var a {{ $TJacobian }}
+	a.ScalarMultiplication(&{{.PointName}}Gen, big.NewInt(42))
+
+	b.Run("equal", func(b *testing.B) {
+		var scalarSquared {{ .CoordType}}
+		scalarSquared.Square(&scalar)
+
+		aZScaled := a
+		aZScaled.X.Mul(&aZScaled.X, &scalarSquared)
+		aZScaled.Y.Mul(&aZScaled.Y, &scalarSquared).Mul(&aZScaled.Y, &scalar)
+		aZScaled.Z.Mul(&aZScaled.Z, &scalar)
+
+		// Check the setup.
+		if !a.Equal(&aZScaled) {
+			b.Fatalf("invalid test setup")
+		}
```

Comment:
Building an equivalent point in jac coordinates, and double-checking that we'll benchmark what we think we'll benchmark.